app/scripts/directives/createSecret.js

```diff
@@ -0,0 +1,177 @@
+              }
+              break;
+            case "kubernetes.io/dockerconfigjson":
+              var encodedConfig = window.btoa(data.dockerConfig);
+              if (JSON.parse(data.dockerConfig).auths) {
+                secret.data[".dockerconfigjson"] = encodedConfig;
+              } else {
+                secret.type = "kubernetes.io/dockercfg";
+                secret.data[".dockercfg"] = encodedConfig;
+              }
+              break;
+            case "kubernetes.io/dockercfg":
+              var auth = window.btoa(data.dockerUsername + ":" + data.dockerPassword);
+              var configData = {};
+              configData[data.dockerServer] = {
+                username: data.dockerUsername,
+                password: data.dockerPassword,
+                email: data.dockerMail,
+                auth: auth
+              };
+              secret.data[".dockercfg"] = window.btoa(JSON.stringify(configData));
+              break;
+          }
+          return secret;
+        };
+
+        var linkSecretToServiceAccount = function(secret) {
+          var updatedSA = angular.copy($scope.serviceAccounts[$scope.newSecret.pickedServiceAccountToLink]);
+          if ($scope.newSecret.linkAs.secrets) {
+            updatedSA.secrets.push({name: secret.metadata.name});
+          }
+          if ($scope.newSecret.linkAs.imagePullSecrets) {
+            updatedSA.imagePullSecrets.push({name: secret.metadata.name});
+          }
+          DataService.update('serviceaccounts', $scope.newSecret.pickedServiceAccountToLink, updatedSA, $scope).then(function(sa) {
+            $scope.alerts["createAndLink"] = {
+              type: "success",
+              message: "Secret " + secret.metadata.name + " was created and linked with service account " + sa.metadata.name + "."
+            };
+          }, function(result){
+            $scope.alerts["createAndLink"] = {
+              type: "error",
+              message: "An error occurred while linking the secret with service account.",
+              details: $filter('getErrorDetails')(result)
+            };
+          });
+        };
+
+        $scope.create = function() {
+          var newSecret = constructSecretObject($scope.newSecret.data, $scope.newSecret.authType);
+          DataService.create('secrets', null, newSecret, $scope).then(function(secret) { // Success
+            if ($scope.newSecret.pickedServiceAccountToLink) {
+              linkSecretToServiceAccount(secret);
+            } else {
+              $scope.alerts["create"] = {
+                type: "success",
+                message: "Secret " + secret.metadata.name + " was created."
+              };
+            }
+            $scope.postCreateAction({newSecret: secret});
+          }, function(result) { // Failure
+            var data = result.data || {};
+            if (data.reason === 'AlreadyExists') {
+              $scope.nameTaken = true;
+            }
```

Comment:
Not sure how, cause the alerts, in case of creating the secrets through the modal will be displayed in the alerts for the BC edit page and not the modal and also not sure if displaying other errors in similar way we do with the name collision(under the input field) is the right thing to do.
Any ideas ?